test/system/505-networking-pasta.bats

```diff
@@ -0,0 +1,675 @@
+	run_podman run --net=pasta $IMAGE ip -j -6 address show
+
+	output="$(ipv6_get_addr_global "${output}")"
+	assert "$(ipv6_get_addr_global)" "Container address not matching host"
+}
+
+@test "podman networking with pasta(1) - IPv6 address assignment" {
+	run_podman run --net=pasta:-a,2001:db8::1 $IMAGE ip -j -6 address show
+
+	output="$(ipv6_get_addr_global "${output}")"
+	assert "2001:db8::1" "Container address not matching configured value"
+}
+
+@test "podman networking with pasta(1) - No IPv6" {
+	skip_if_no_ipv4
+
+	run_podman run --net=pasta:-4 $IMAGE ip -j -6 address show
+
+	output="$(ipv6_get_addr_global "${output}")"
+	assert "null" "Container has IPv6 global unicast with IPv6 disabled"
+}
+
+### Routes #####################################################################
+
+@test "podman networking with pasta(1) - IPv4 default route" {
+	run_podman run --net=pasta $IMAGE ip -j -4 route show
+
+	output="$(ipv4_get_route_default "${output}")"
+	assert "$(ipv4_get_route_default)" "Container route not matching host"
+}
+
+@test "podman networking with pasta(1) - IPv4 default route assignment" {
+	run_podman run --net=pasta:-a,192.0.2.2,-g,192.0.2.1 $IMAGE \
+		ip -j -4 route show
+
+	output="$(ipv4_get_route_default "${output}")"
+	assert "192.0.2.1" "Container route not matching configured value"
+}
+
+@test "podman networking with pasta(1) - IPv6 default route" {
+	run_podman run --net=pasta $IMAGE ip -j -6 route show
+
+	output="$(ipv6_get_route_default "${output}")"
+	assert "$(ipv6_get_route_default)" "Container route not matching host"
+}
+
+@test "podman networking with pasta(1) - IPv6 default route assignment" {
+	run_podman run --net=pasta:-a,2001:db8::2,-g,2001:db8::1 $IMAGE \
+		ip -j -6 route show
+
+	output="$(ipv6_get_route_default "${output}")"
+	assert "2001:db8::1" "Container route not matching configured value"
+}
+
+### Interfaces #################################################################
+
+@test "podman networking with pasta(1) - Default MTU" {
+	run_podman run --net=pasta $IMAGE ip -j link show
+
+	output="$(ether_get_mtu "${output}")"
+	assert "65520" "Container's default MTU not 65220 bytes by default"
+}
+
+@test "podman networking with pasta(1) - MTU assignment" {
+	run_podman run --net=pasta:-m,1280 $IMAGE ip -j link show
+
+	output="$(ether_get_mtu "${output}")"
+	assert "1280" "Container's MTU not matching configured 1280 bytes"
+}
+
+@test "podman networking with pasta(1) - Loopback interface state" {
+	run_podman run --net=pasta $IMAGE ip -j link show
+
+	local jq_expr='.[] | select(.link_type == "loopback").flags | '\
+'		       contains(["UP"])'
+
+	output="$(printf '%s' "${output}" | jq -rM "${jq_expr}")"
+	assert "true" "Container's loopback interface not up"
+}
+
+### DNS ########################################################################
+
+@test "podman networking with pasta(1) - External resolver, IPv4" {
+	run_podman run --net=pasta $IMAGE nslookup 127.0.0.1
```

Comment:
I'm not sure if I should be screen-scraping from `nslookup`: the output might change across Busybox versions, and is definitely different between Busybox and the ISC (BIND 9) version. For example, Busybox 1.35.0 might report:

```
Server:		185.12.64.1
Address:	185.12.64.1:53

1.0.0.127.in-addr.arpa	name = localhost

```

whereas ISC's nslookup 9.16.33 would say:
```
1.0.0.127.in-addr.arpa	name = localhost.

```
The return code however is quite universal, in the sense that `0` indicates a successful resolution.

I also don't understand two things, specifically:
* which command line are we talking about? That is, what would evaluate to `echo`?
* if a "bad" `nslookup` results in a wrong error code, wouldn't that be the case for the output, too?

Reply:
I think it's safe to check `assert "$output" =~ "127\.in-addr\.arpa"`. That's a pretty solid guarantee that nslookup ran and resolved.

As for `echo`, I'm talking general test-writing principles. It's super common for a test writer to (temporarily) disable a test by doing something like `run_podman ... echo nslookup blah`, and forget to reenable it. A reviewer is likely to miss tiny things like that, but not likely to miss a large block of commented-out or disabled checks. 

As for broken nslookup, a future update to `$IMAGE` could potentially include an `nslookup` that somehow just NOPs on certain args, or otherwise exits success without doing the request. Tests must be written to check output, and more importantly, output that is not a simple substring-check of the given input. (like, for `run_podman nslookup 127.0.0.1`, you never check for `127.0.0.1`. You check for a transformation of it, one that is impossible via just `echo`).

Reply:
> As for `echo`, I'm talking general test-writing principles. It's super common for a test writer to (temporarily) disable a test by doing something like `run_podman ... echo nslookup blah`, and forget to reenable it. A reviewer is likely to miss tiny things like that, but not likely to miss a large block of commented-out or disabled checks.

To me, that rather sounds like a (worrying) weakness in the process.

> As for broken nslookup, a future update to `$IMAGE` could potentially include an `nslookup` that somehow just NOPs on certain args, or otherwise exits success without doing the request.

Well, it worked like that for about 36 years that BIND is around... but anyway,

> Tests must be written to check output, and more importantly, output that is not a simple substring-check of the given input. (like, for `run_podman nslookup 127.0.0.1`, you never check for `127.0.0.1`. You check for a transformation of it, one that is impossible via just `echo`).

...sure, I'll change it to your version, I hope it's robust enough.

I think it would be helpful to have these guidelines spelled out in the tests README, because they appear to be very specific. I have some experience with other FLOSS test frameworks (mostly kernel selftests, and nftables), I occasionally review submissions and I happened to mentor on related projects: the general principle there is to clearly establish what stable interfaces are, before writing test cases. Human-readable output is never considered one of them (well, it's in its name). Return codes are. JSON output generally is, for tools that support it.

Reply:
Done.